src/darwin/Framework/CHIP/MTRDeviceControllerDelegateBridge.mm

```diff
@@ -107,5 +131,13 @@
                 [strongDelegate controller:strongController commissioningComplete:nsError];
             });
         }
+
+        if ([strongDelegate respondsToSelector:@selector(controller:commissioningComplete:deviceId:)]) {
```

Comment:
Shouldn't we only dispatch one of the two notifications?

Reply:
Yes, that’s right. However, since it is a strongDelegate, it may not be delivered if it is not defined in the app, or only one may be delivered.

Reply:
`strongDelegate respondsToSelector:@selector`
As you may know, this code is called only when it is implemented in the App Layer. This was implemented because there are cases where existing developer use commissioningComplete without a deviceId.

Reply:
I understand why the new thing is being added, yes.  My question, again, is: If the delegate implements both selectors, why are we invoking both instead of just the "new" one?  Please don't resolve this conversation; I will resolve it once that question is answered.

Reply:
I think that such a case does not exist.

Reply:
What do you mean by "such a case does not exist"?  I described above, in https://github.com/project-chip/connectedhomeip/pull/25832#discussion_r1149292388 why a delegate would want to implement both selectors.

Reply:
I'd like to cut it down to one, but I've left it out for old API users.
Would it be more appropriate to mark the previous implementation as "MTR_DEPRECATED"?

Reply:
That's fine (and yes, MTR_DEPRECATED is probably good to add), but my point is the logic should check for the new selector, if supported call that, if not then check for the old selector, etc.

Reply:
I updated. (Add return, MTR_DEPRECATED)